### ISSUE TITLE
Updating footnotes

### ISSUE THREAD

**Issue by mgifford** [wet-boew/wet-boew#8826]
I was looking at the footnotes on this page here:

https://www.canada.ca/en/revenue-agency/programs/about-canada-revenue-agency-cra/income-statistics-gst-hst-statistics/disability-tax-credit-statistics/disability-tax-credit-glance.html

I’m a bit baffled as to why someone would use aria-labelledby in a description list, since it is kinda redundant:

```
<dt id="fn1-dt">Footnote 1</dt>
<dd id="fn1" tabindex="-1" aria-labelledby="fn1-dt">Stuff…</dd>
```

Going back to [rule 1](https://www.deque.com/blog/top-5-rules-of-aria/) of ARIA it is a bad idea.
Also because there is a good case to semantically link the footnote definition to the table heading that it is describing. If ARIA was being used to semantically link the footnote to the table heading that that would be a different thing (and probably quite useful). 

This comes right from https://wet-boew.github.io/v4.0-ci/demos/footnotes/footnotes-en.html

And here:
https://github.com/wet-boew/wet-boew/blob/master/src/plugins/footnotes/footnotes.js#L39

But the example Footnote 
https://wet-boew.github.io/v4.0-ci/demos/footnotes/footnotes-en.html#fn

and the code aren't the same:
https://wet-boew.github.io/v4.0-ci/demos/footnotes/footnotes-en.html#wb-auto-2

In WET5 it might be better to do this with Tooltips.  Footers make a lot of sense for a pattern on a printed page, but don't necessarily help you understand a table or other complex image if you're having to jump down then scroll back up to the table in question. 

https://github.com/wet-boew/wet-boew/issues/5273

**Comment by EricDunsworth** [comment]
FWIW ``aria-labelledby`` was first introduced into WET 3.1's incarnation of the footnotes plugin via wet-boew/wet-boew-legacy@35c3e0e735ba0f309e1d5b321df3ce136640a42f. It resolved #361. Although that issue wasn't specifically asking to add that attribute.

**Excerpts from #361...**

**@EricDunsworth https://github.com/wet-boew/wet-boew/issues/361#issuecomment-8345788:**
> **2)**
> Do you think the change involving setting IDs on DDs (instead of DTs) may have any beneficial side effects for screen readers (like not needing to read through a DT called "Footnote #" after accessing a link already labelled as "Footnote #")?
> 
> I haven't gotten around to sending a pull request from my own footnotes repository (which has some additional updates - such as much improved print CSS support, better highlighting effects, and more) just yet, but in older versions of that repository I was achieving highlighting effects via :target/:focus and adjacent sibling selectors (e.g. `dt:focus + dd {/*dd highlighting properties*/}`. Although in hindsight, the way in which I was going about it may not have worked out very well in scenarios where a given footnote's content contained links... IE7/8 also weren't playing nicely with what I was doing before.

**@pjackson28 https://github.com/wet-boew/wet-boew/issues/361#issuecomment-8346383:**
> 2. It might of but just in case it was still useful I created an association back to the dt with aria-labelled by. The real benefit is the visual focus which helps to identify to the user where they are located. The problem with the CSS effects you were playing with wa that they were indicating visual focus on the dd when it wasn't actually the case.

**Comment by mgifford** [comment]
Thanks for the context @EricDunsworth 

Ok, so this might be a good design decision in 2012.  Does it still apply in 2020?

How could it be improved for WET5?

**Comment by EricDunsworth** [comment]
In retrospect, my take on it is that everyone might've been a bit *too* eager to tack ARIA/role attributes onto everything at the time (not just based on necessity). It might've been overdone to some degree.

In this specific case, what was done was perceived as a benefit to screen readers. But I don't know if any research was done to prove its usefulness at the time. Either way, as you mentioned, it's 8 years later now and the HTML spec seems pretty clear about [how user agents are meant to establish the relationships between ``<dt>`` and ``<dd>`` elements](https://html.spec.whatwg.org/multipage/grouping-content.html#the-dl-element). So I don't see any reason as to why a well-designed modern-day screen reader wouldn't be able to infer those relationships based on the native semantics.

IMO it'd be up to the AWG (@bsouster?) to evaluate it and make a determination about whether it'd be best to keep it or scrap it.

Btw the reason ``aria-describedby`` wasn't hardcoded into the footnotes plugin's code examples was probably because https://github.com/wet-boew/wet-boew-legacy/commit/35c3e0e735ba0f309e1d5b321df3ce136640a42f was introduced in a WET 3.1.x maintenance release. So it was implemented as a JS-driven progressive enhancement to avoid introducing HTML markup changes.

**Comment by bsouster** [comment]
@EricDunsworth,
@grifare and I will do test cases with real users to see if the ARIA currently adds anything of value

**Comment by grifare** [comment]
Here is the list of my initial findings:

- With or without aria, user agents do not establish any relationship between `<dt>` and `<dd>` elements. In other words, descriptions follow terms and users are expected to associate them naturally . I personally think this is not asking for too much: I could very easily associate a description that followed the term, by listening only.
This explanation is from WCAG:
> The list is marked up using the dl element. Within the list, each term is put in a separate dt element, and **its description goes in the dd element directly following it**. Multiple terms can be associated with a single description, as can a single term with multiple descriptions, **provided that semantic sequence is maintained**.
[w3.org/TR/WCAG20-TECHS/H40.html](https://www.w3.org/TR/WCAG20-TECHS/H40.html)

This code does not need aria to tell a blind user milk is not a black hot drink.
```
<dl title="drinks">
  <dt>Coffee</dt>
  <dd>Black hot drink</dd>
  <dt>Milk</dt>
  <dd>White cold drink</dd>
</dl>
```
- I think **aria-describedby or aria-labelledby work on interactive elements**. My understanding is that, when some retrospective fixes are required, aria is intended to function as a `<label>` element with a properly-coded for attribute. In other words, aria does not make screen readers say anything more than what is visible in static content. For example a `<p>` element cannot simply become an explanation for another `<p>` element just because an aria-describedby is used. The aria code below does nothing.
```
<p aria-describedby="milk">Milk</p>
<p id="milk">White cold drink</p>
```
- In the [example WCAG gave](https://www.w3.org/TR/WCAG20-TECHS/H40.html), `<dt>` and `<dd>` elements are associated manually by the author. Please note the repeated usage of the terms that are first provided in `<dt>` elements, in `<dd>` elements.  WCAG would have suggested a programmatic solution, if there were one.
```
<dl title="Nautical terms">
  <dt>Knot</dt>
  <dd>
    <p>A <em>knot</em> is a unit of speed equaling 1 
      nautical mile per hour (1.15 miles per hour or 1.852 
      kilometers per hour).</p>
  </dd>
  <dt>Port</dt>
  <dd>
    <p><em>Port</em> is the nautical term (used on 
      boats and ships) that refers to the left side
      of a ship, as perceived by a person facing towards 
      the bow (the front of the vessel).</p>
  </dd>
  <dt>Starboard</dt>
  <dd>
    <p><em>Starboard</em> is the nautical term (used 
      on boats and ships) that refers to the right 
      side of a vessel, as perceived by a person 
      facing towards the bow (the front of the vessel).</p>
  </dd>
</dl>
```
- I am afraid the functionality of the majority of the aria codes have never been actually verified, neither before HTML5 nor after it. As @mgifford put at the beginning, the [rule #1](https://www.deque.com/blog/top-5-rules-of-aria/) of aria is all too often ignored.

A real user user testing will follow.

**Comment by grifare** [comment]
Here is the code tested:
```
<!DOCTYPE html>
<html lang="en">
<head>
<title>Data Lists</title>
</head>
<body>
	<h1>HTML5 Data Lists</h1>
	<h2>Data list with no ARIA</h2>
	<dl title="drinks">
	  <dt>Coffee</dt>
	  <dd>Black hot drink</dd>
	  <dt>Milk</dt>
	  <dd>White cold drink</dd>
	</dl>
	
	<h2>The same data list with ARIA labelled by</h2>
	<dl title="drinks">
	  <dt id="coffee">Coffee</dt>
	  <dd aria-labelledby="coffee">Black hot drink</dd>
	  <dt id="milk">Milk</dt>
	  <dd aria-labelledby="milk">White cold drink</dd>
	</dl>
	
	<h2>The same data list with ARIA described by</h2>
	<dl title="drinks">
	  <dt aria-describedby="coffee">Coffee</dt>
	  <dd id="coffee">Black hot drink</dd>
	  <dt aria-describedby="milk">Milk</dt>
	  <dd id="milk">White cold drink</dd>
	</dl>
	<aside>End of file</aside>
</body>
</html>
```
**Using JAWS, the result of the test is that ARIA does nothing here**.
I shared the [audio recording of the screen reader NVDA](https://drive.google.com/open?id=1yPeK1T0-tsnUFbJwExMppqaxORyRBzXr) on Google Drive. (Sorry for the poor quality, could not professionally record it.)

**Comment by bsouster** [comment]
I will add it as an item for discussion next Access Working Group (AWG) meeting

**Comment by shawnthompson** [comment]
I would remove it but I still think footnotes on the web would do better if they were presented in a dialog box (modal, lightbox, overlay) or an accessible tooltip. Keeping in mind they should show up when printed. For all those environmental haters out there.

**Comment by dfarough** [comment]
At least we do not get extra verbosity.
but given the testing and results provided here, I think it should be removed since clearly it provides no value and may cause people to use this technique when it is not useful.
